dask/base.py

```diff
@@ -139,11 +141,20 @@ def normalize_function(func):
 
 
 normalize_token = Dispatch()
-normalize_token.register((int, float, str, tuple, list), lambda a: a)
-normalize_token.register(object,
-        lambda a: normalize_function(a) if callable(a) else a)
+normalize_token.register((int, float, str, unicode, bytes, tuple, list,
```

Comment:
What happens if you pass in a list or tuple of arbitrary objects as args? I think you need to call `normalize_token` recursively?